bdk_chain/src/chain_graph.rs

```diff
@@ -39,6 +39,24 @@ impl<P, T> Default for ChainGraph<P, T> {
     }
 }
 
+impl<P, T> AsRef<SparseChain<P>> for ChainGraph<P, T> {
+    fn as_ref(&self) -> &SparseChain<P> {
+        &self.chain
+    }
+}
+
+impl<P, T> AsRef<TxGraph<T>> for ChainGraph<P, T> {
+    fn as_ref(&self) -> &TxGraph<T> {
+        &self.graph
+    }
+}
+
+impl<'a, P, T> AsRef<ChainGraph<P, T>> for &'a ChainGraph<P, T> {
```

Comment:
Ah so this is the PR its from. 
```suggestion
impl<P, T> AsRef<ChainGraph<P, T>> for ChainGraph<P, T> {
```
is what you want.

Reply:
Does that work? Do you mean:
```rust
impl<P, T> AsRef<ChainGraph<P, T>> for &ChainGraph<P, T> { .. }
```

Reply:
Oh damn, it works